primitives/rpc/debug/src/api/mod.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright 2019-2021 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Runtime API allowing to debug/trace Ethereum
+
+use codec::{Decode, Encode};
+use ethereum::TransactionV0 as Transaction;
+use ethereum_types::H160;
+use sp_std::vec::Vec;
+
+#[cfg(feature = "std")]
+use crate::serialization::*;
+#[cfg(feature = "std")]
+use serde::Serialize;
+
+pub mod block;
+pub mod single;
+
+pub const V2_RUNTIME_VERSION: u32 = 400;
+
+sp_api::decl_runtime_apis! {
+	#[api_version(3)]
+	pub trait DebugRuntimeApi {
```

Comment:
Wondering, old runtime api versions that use this trait will not work now that has been moved right?

Reply:
Hmm, I would guess it use the name of the trait and its methods, but maybe it uses its path. In that case indeed it will not work.

But yeah I assume we will transition to using substitutes for all tracing.

Reply:
If we are planning to build the client that uses substitutes on top of this, maybe we can get remove the runtime api versions (and the host function versions), and build the runtimes with a single version. The only main difference we need to keep is the manual initialization of blocks for new substrate versions, and assume they are initialized for old substrate versions. But that can be done client side.